sklearn/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,312 @@
+
+    Parameters
+    ----------
+
+    non_default : type
+        A parameter with no default initialization.
+
+    default : type
+        A parameter with a default initialization.
+
+    default_2 : type
+        Another paramter with a default initialization.
+
+    additional : type
+        The additional parameter which is not present in argspec.
+    """
+
+
+def f4(non_default, default=1, default_2=None, *args):
+    """A test function just an additional parameter in docstring.
+    
+    The default values in argspec initialized to None are not considered.
+
+    Parameters
+    ----------
+    foo : type
+    """
+
+
+def f5(non_default, default=1, *args, **kwargs):
+    """A test function with both additional parameter in docstring and argspec.
+    
+    Presence of kwargs makes the additional parameter(s) to not show up
+    as mismatch. Only `default` is shown as missing from docs.
+
+    Parameters
+    ----------
+
+    non_default : type
+        A parameter with no default initialization.
+
+    default_2 : type
+        Another paramter with a default initialization.
+
+    additional : type
+        The additional parameter which is not present in argspec.
+    """
+
+
+def f6(non_default, default=1, default_2=None, *args):
+    """A test function just an additional parameter in docstring.
+    
+    The default values in argspec initialized to None are not considered.
+
+    Parameter
+    ---------
+    foo : type
+    """
+
+
+def get_name(func):
+    parts = []
+    module = inspect.getmodule(func)
+    if module:
+        parts.append(module.__name__)
+    if hasattr(func, 'im_class'):
+        parts.append(func.im_class.__name__)
+    parts.append(func.__name__)
+    return '.'.join(parts)
+
+
+def check_parameters_match(func, doc=None, debug=False):
+    result = Counter({'docs':0, 'argspec':0, 'warnings':0})
+
+    name = get_name(func)
+
+    if not debug:
+        if not name.startswith('sklearn.'):
+            return result
+        if inspect.isdatadescriptor(func):
+            return result
+
+        # Ignore abstract methods.
+        source = inspect.getsource(func).strip('\n ')
+        if source.endswith('\"\"\"') or source.endswith('\'\'\'') or \
+                source.endswith('pass'):
+            return result
+
+    try:
+        args, varargs, varkw, defaults = inspect.getargspec(func)
+    except TypeError:
+        return result
+
+    if 'self' in args:
+        args.remove('self')
+
+    # To capture warnings related to invalid section names in docs
+    warnings.simplefilter('always')
+    if doc is None:
+        with warnings.catch_warnings(record=True) as w:
+            try:
+                doc = docscrape.FunctionDoc(func)
+            except ValueError:
+                print('Failed to parse doc of %r' % func, file=sys.stderr)
+                return result
+            msg = ('Warning was raised when attempting to scrape the doc '
+                   'of the class with name %s.\n'
```

Comment:
I added this so that one can find the class whose docstring has invalid section name(s)...
The warning message will simply state that "Parameter is an unknown section" without any other traceback...